controllers/healthcheck_controller.go

```diff
@@ -1184,39 +1263,47 @@ func (r *HealthCheckReconciler) createRemedyNameSpaceRole(clientset kubernetes.I
 		},
 	})
 	if err != nil {
-		return "", err
+		return "", false, err
 	}
 
-	return nsrole1.Name, nil
+	return nsrole1.Name, true, nil
 }
 
 // Delete NamespaceRole
-func (r *HealthCheckReconciler) DeleteNameSpaceRole(clientset kubernetes.Interface, nsrole string, namespace string) error {
+func (r *HealthCheckReconciler) DeleteNameSpaceRole(clientset kubernetes.Interface, nsrole string, namespace string) (bool, error) {
 	// Check if a Namespace Role already exists
-	_, err := clientset.RbacV1().Roles(namespace).Get(nsrole, metav1.GetOptions{})
+	nr, err := clientset.RbacV1().Roles(namespace).Get(nsrole, metav1.GetOptions{})
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	err = clientset.RbacV1().Roles(namespace).Delete(nsrole, &metav1.DeleteOptions{})
-	if err != nil {
-		return err
+	// Delete only if created by the workflow
+	if val, ok := nr.Labels[WfManagedByLabelKey]; ok && val == WfManagedByValue {
+		err = clientset.RbacV1().Roles(namespace).Delete(nsrole, &metav1.DeleteOptions{})
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+
 	}
 
-	return nil
+	return false, nil
 }
 
 // Create a NamespaceRoleBinding
-func (r *HealthCheckReconciler) createNameSpaceRoleBinding(clientset kubernetes.Interface, rolebinding string, nsrole string, serviceaccount string, namespace string) (string, error) {
+func (r *HealthCheckReconciler) createNameSpaceRoleBinding(clientset kubernetes.Interface, rolebinding string, nsrole string, serviceaccount string, namespace string) (string, bool, error) {
```

Comment:
instead of returning a boolean update the same function here with logic above:
```
			if <True> {
				log.Info("Successfully Created", "NamespaceRoleBinding", nsrb)
				r.Recorder.Event(hc, v1.EventTypeNormal, "Normal", "Successfully Created NamespaceRoleBinding")
			} else {
				log.Info("Found existing", "NamespaceRoleBinding", nsrb)
				r.Recorder.Event(hc, v1.EventTypeNormal, "Normal", "Existing NamespaceRoleBinding found")
			}
```

Reply:
made the changes.